pr_toolbox/quantum/results/reckoning.py

```diff
@@ -210,11 +206,10 @@ def _validate_frequencies_list(
     @staticmethod
     def _validate_frequencies(frequencies: FrequenciesLike) -> QuasiDistribution:
         """Validate frequencies."""
-        if isinstance(frequencies, QuasiDistribution):
-            return frequencies
-        if isinstance(frequencies, (Counts, dict)):
+        if isinstance(frequencies, dict) and not isinstance(frequencies, QuasiDistribution):
```

Comment:
You are right about this one, let's keep your original version and simply add a comment about why we need the check.
```suggestion
        if isinstance(frequencies, QuasiDistribution):
            return frequencies  # Note: issubclass(QuasiDistribution, dict) is True
        if isinstance(frequencies, dict):
```